tests/unittests/test_datasource/test_ec2.py

```diff
@@ -692,10 +847,36 @@ def test_convert_ec2_metadata_network_config_handles_local_v4_and_v6(self):
         nic1_metadata['ipv6s'] = '2620:0:1009:fd00:e442:c88d:c04d:dc85/64'
         nic1_metadata.pop('public-ipv4s')
         nic1_metadata['local-ipv4s'] = '10.0.0.42'  # Local ipv4 only on vpc
-        expected = {'version': 1, 'config': [
-            {'mac_address': self.mac1, 'type': 'physical',
-             'name': 'eth9',
-             'subnets': [{'type': 'dhcp4'}, {'type': 'dhcp6'}]}]}
+        expected = {'version': 2, 'ethernets': {'eth9': {
+            'match': {'macaddress': self.mac1.lower()}, 'set-name': 'eth9',
+            'dhcp4': True, 'dhcp4-overrides': {'route-metric': 100},
+            'dhcp6': True, 'dhcp6-overrides': {'route-metric': 100}}}}
+        self.assertEqual(
+            expected,
+            ec2.convert_ec2_metadata_network_config(
+                network_metadata_both, macs_to_nics))
+
+    def test_convert_ec2_metadata_network_config_handles_multiple_nics(self):
+        """When dhcp6 is public and dhcp4 is set to local enable both."""
```

Comment:
This comment still needs clarification;  maybe drop it;  we just want to see that for multi nic, we dhcp4 on both interfaces, dhcp6 only on interfaces which have ipv6s non-empty; and all secondary ips (v4 and v6) are present.

Reply:
Changed to
        """When ipv6s and local-ipv4s are non-empty, enable dhcp6 and dhcp4."""